### PR TITLE
Implement deleteAllById() in SimpleDatastoreRepository

### LINKED ISSUE #565: Implement in deleteAllById() in `SimpleDatastoreRepository`
Spring Data 2.5 introduced a new method in `CrudRepository`: [`deleteAllById()`](https://github.com/spring-projects/spring-data-commons/pull/476). Since we support both Spring Boot 2.4 and 2.5, this means we'll have to omit `@Overrides` annotation to maintain ability to compile against 2.4.

### PR DESCRIPTION
**Pull request by zhumin8**
Implement deleteAllById() in SimpleDatastoreRepository and add relevant tests.
Resolves #565

### CODE REVIEW

File: spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepositoryTests.java [hunk 1]
```diff
@@ -353,8 +352,9 @@ public void findAllSortAsc() {
 	}
 
 	@Test
-	public void deleteAllByIdUnimplemented() {
-		assertThatThrownBy(() -> this.simpleDatastoreRepository.deleteAllById(new ArrayList<>()))
-				.isInstanceOf(UnsupportedOperationException.class);
+	public void deleteAllById() {
+		List<String> keys = Arrays.asList("1", "2");
+		this.simpleDatastoreRepository.deleteAllById(keys);
+		verify(this.datastoreTemplate).deleteAllById(keys, Object.class);
```
Comment (zhumin8) [inline spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepositoryTests.java:358]: This unit test only verifies `deleteAllById` of datastoreTemplate mock is called.
Test for the actual methods is already tested in [DatastoreTemplateTests](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplateTests.java#L923-L938)

File: spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java [hunk 2]
```diff
@@ -544,6 +546,17 @@ public void testDelete() {
 		assertThat(this.testEntityRepository.findById(1L)).isNotPresent();
 	}
 
+	@Test
+	public void deleteAllByIdTest() {
+		assertThat(this.testEntityRepository.findAllById(Arrays.asList(1L, 2L))).hasSize(2);
+		// cast to SimpleDatastoreRepository for method be reachable with Spring Boot 2.4
+		SimpleDatastoreRepository simpleRepository = AopTestUtils
```
Comment (zhumin8) [inline spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java:553]: `this.testEntityRepository` is a Spring Proxy object, and could not cast it to `SimpleDatastoreRepository` directly. Found this utility method to unwrap it.

**Review by meltsufin — APPROVED** [review]
LGTM.
Do we also need to implement the same for Spanner and Firestore?

File: spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java (hunk 2 above)
Comment (elefeint) [reply spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java:553]: oh, this is very cool!

File: spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java [hunk 3]
```diff
@@ -544,6 +546,17 @@ public void testDelete() {
 		assertThat(this.testEntityRepository.findById(1L)).isNotPresent();
 	}
 
+	@Test
+	public void deleteAllByIdTest() {
+		assertThat(this.testEntityRepository.findAllById(Arrays.asList(1L, 2L))).hasSize(2);
```
Comment (elefeint) [inline spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java:551]: Nice precondition check.

File: spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepositoryTests.java (hunk 1 above)
Comment (elefeint) [reply spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepositoryTests.java:358]: That's good, even if it feels weird. It's a unit test that only tests the functionality that you changed in this class.
You've added an integration test to validate end-to-end functionality, so it's completely fair to only have the unit test do this much.

**Review by elefeint — APPROVED** [review]
Did not need to be a draft, after all. I have no comments, and I learned something new!

**Comment by elefeint** [comment]
@meltsufin We do; there are separate tracking issues for both, and a checklist in #472.

**Pull request merged**